src/plugins/inputs/ignore.js

```diff
@@ -110,35 +111,41 @@ exports.input = function (network, chan, cmd, args) {
 					})
 				);
 			} else {
-				const chanName = "Ignored users";
-				const ignored = network.ignoreList.map((data) => ({
-					hostmask: `${data.nick}!${data.ident}@${data.hostname}`,
-					when: data.when,
-				}));
-				let newChan = network.getChannel(chanName);
-
-				if (typeof newChan === "undefined") {
-					newChan = client.createChannel({
-						type: Chan.Type.SPECIAL,
-						special: Chan.SpecialType.IGNORELIST,
-						name: chanName,
-						data: ignored,
-					});
-					client.emit("join", {
-						network: network.uuid,
-						chan: newChan.getFilteredClone(true),
-						index: network.addChannel(newChan),
-					});
-				} else {
-					newChan.data = ignored;
-
-					client.emit("msg:special", {
-						chan: newChan.id,
-						data: ignored,
-					});
-				}
+				create_new_ignored_window = true;
 			}
 
 			break;
 	}
+
+	const chanName = "Ignored users";
```

Comment:
Now, suddenly ignore + unignore end up here, even though you already know that they shouldn't.
The only code path that should ever see this is ignorelist, so scope it to that.
If it gets too long to inline it into the switch (which may very well be the case), write a function for it.